bundles/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/se/cache/ValloxExpiringCacheMap.java

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.vallox.internal.se.cache;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.vallox.internal.se.mapper.ChannelDescriptor;
+import org.openhab.binding.vallox.internal.se.telegram.Telegram;
+
+/**
+ * Expiring cache implementation.
+ *
+ * @author Miika Jukka - Initial contribution
+ */
+@NonNullByDefault
+public class ValloxExpiringCacheMap {
```

Comment:
I'm familiar with it and did consider it some time ago and thought about extending it to include some of the methods that `ValloxExpiringCacheMap` has. As I imagine implementing it we would need some abstraction layer to fit it in. For example the exceptions need handling. So came to the conclusion that it simpler and suits the needs better this way. But don't assume that I'm against it, I just need to be convinced otherwise.

Reply:
Ok, let's leave this as it is and maybe in the future I'll do the change. Would be better to use something already in the framwork.